src/lib.rs

```diff
@@ -124,6 +124,13 @@ impl XorName {
         XOR_NAME_BITS
     }
 
+    /// Returns a copy of `self`, with the `i`-th bit flipped.
+    pub fn with_flipped_bit(&self, index: usize) -> XorName {
+        let &XorName(mut bytes) = self;
+        bytes[index / 8] = bytes[index / 8] ^ (1 << (7 - index % 8));
```

Comment:
Well, I guess as a tie breaker, safety should always beat convenience, so I grudgingly made it a `Result`. :-)